src/formatting/expr.rs

```diff
@@ -125,7 +125,31 @@ pub(crate) fn format_expr(
         | ast::ExprKind::While(..) => to_control_flow(expr, expr_type)
             .and_then(|control_flow| control_flow.rewrite(context, shape)),
         ast::ExprKind::ConstBlock(ref anon_const) => {
-            Some(format!("const {}", anon_const.rewrite(context, shape)?))
+            let between_span = mk_sp(
+                context.snippet_provider.span_after(expr.span, "const"),
+                anon_const.value.span.lo(),
+            );
+            let anon_const_str = anon_const.rewrite(context, shape)?;
+            let combined_strs = combine_strs_with_missing_comments(
+                context,
+                "const",
+                &anon_const_str,
+                between_span,
+                shape,
+                false,
+            )?;
+
+            match context.config.brace_style() {
+                BraceStyle::AlwaysNextLine => Some(format!("{}", &combined_strs)),
+                _ => {
+                    let comment = recover_missing_comment_in_span(between_span, shape, context, 0)?;
+                    if comment.is_empty() {
+                        Some(format!("const {}", &anon_const_str))
+                    } else {
+                        Some(format!("{}", &combined_strs))
+                    }
+                }
```

Comment:
> e.g. this is fine when brace_style is not AlwaysNextLine on these branches
correct, currently it will add a new line prior to the brace.

Reply:
Right, and though we want to adjust that long term, we don't have to do that here. `ConstBlock` is still relatively new syntax so we have the bandwidth to iterate on this a bit. Since this PR enhances things so that rustfmt will be able to handle `ConstBlock`'s with comments, I'm comfortable merging it even though we'll need to follow up with a replacement for this logic to continue to improve the formatting of comments in this case.

Reply:
Fixed as you suggest.
i mark this as resolve to let the discussion on track with the new update.